addons/hr/models/hr_employee_base.py

```diff
@@ -89,6 +94,11 @@ def _search_newly_hired(self, operator, value):
         op = 'in' if value and operator == '=' or not value and operator != '=' else 'not in'
         return [('id', op, new_hires.ids)]
 
+    @api.depends('work_location_id')
+    def _compute_type_name_work_location_display(self):
+        for employee in self:
+            employee.name_work_location_display = employee.work_location_id.name if employee.work_location_id else _('Unspecified')
+            employee.type_work_location = employee.work_location_id.location_type if employee.work_location_id else 'other'
```

Comment:
Simpler way of writing it (also handling better fallback if the name or location_type is somehow empty)
```suggestion
            employee.name_work_location_display = employee.work_location_id.name or _('Unspecified')
            employee.type_work_location = employee.work_location_id.location_type or 'other'
```